src/ui/public/agg_table/agg_table.js

```diff
@@ -96,22 +99,22 @@ uiModules
           }
 
           const isFieldNumeric = (field && field.type === 'number');
-          const isFirstValueNumeric = _.isNumber(_.get(table, `rows[0][${i}].value`));
```

Comment:
It would have also matched fields that are not considered numeral by Kibana but contained a number in the first row.  The most common match was for date.  Boolean might have also matched.  I didn't test boolean to see if it shows up as a number or not.
But I worried that if that field wasn't a number, there is a high possibility that other rows will have non numeric values and that would mess up the totals.  I could have it loop through all of the rows and check that they all are numeric if you'd like.